### PR TITLE
removed redundant check in `removeListener`

### PR DESCRIPTION
**Pull request by bekzod**

### CODE REVIEW

File: packages/ember-metal/lib/events.js [hunk 1]
```diff
@@ -126,11 +126,7 @@ export function removeListener(obj, eventName, target, method) {
     target = null;
   }
 
-  metaFor(obj).removeFromListeners(eventName, target, method, (...args) => {
-    if ('function' === typeof obj.didRemoveListener) {
```
Comment (bekzod) [inline packages/ember-metal/lib/events.js]: callback is checked for being `function` inside `meta. removeFromListeners ` already

File: packages/ember-metal/lib/events.js [hunk 2]
```diff
@@ -126,11 +126,7 @@ export function removeListener(obj, eventName, target, method) {
...
-      obj.didRemoveListener(...args);
-    }
-  });
+  metaFor(obj).removeFromListeners(eventName, target, method, obj.didRemoveListener);
```
Comment (stefanpenner) [inline packages/ember-metal/lib/events.js]: this will result in `didRemoveListener` being invoked without a `this`

**Review by stefanpenner — CHANGES_REQUESTED** [review]

File: packages/ember-metal/lib/events.js (hunk 2 above)
Comment (bekzod) [inline packages/ember-metal/lib/events.js]: oops..... I remade what I did.... it is better that before I think

File: packages/ember-metal/lib/events.js [hunk 3]
```diff
@@ -126,11 +126,9 @@ export function removeListener(obj, eventName, target, method) {
     target = null;
   }
 
-  metaFor(obj).removeFromListeners(eventName, target, method, (...args) => {
-    if ('function' === typeof obj.didRemoveListener) {
-      obj.didRemoveListener(...args);
-    }
-  });
+  let func = ('function' === typeof obj.didRemoveListener) ?
+    obj.didRemoveListener.bind(obj) : ()=> {};
```
Comment (bekzod) [inline packages/ember-metal/lib/events.js]: * avoids creating wrapping function
* does not do `if check` each cycle
* does not convert arguments to array
* does not do work when `obj.didRemoveListener` missing

Comment (stefanpenner) [inline packages/ember-metal/lib/events.js]: bind is heavily optimized in modern, this may be a good idea.

**Review by stefanpenner — APPROVED** [review]

**Comment by krisselden** [comment]
we should be using bind, not creating closures.  only reason in the past for not doing it was phantom, I don't think that is a good reason

**Pull request merged**